### DISCUSSION TITLE
Is spending a TapRoot (P2TR) utxo supported?

### DISCUSSION THREAD

**Discussion by paulo-bc** [trustwallet/wallet-core#3936] [Q&A]
Dear friends,

Asking again what was raised in #3296 as I don't see a way to re-open that question.

Is spending a TapRoot (P2TR) utxo supported?
My goal is to be able to 'send/transfer' an ordinal, the transaction would have: one P2TR input; one or more P2WPKH inputs; one P2TR output; one P2WPKH output.

I have been unable to locate a unit test that covers this specific scenario.

Kind regards,
Paulo Lopes

**Comment by weixuefeng** [comment]
not supported now!

**Reply by shreyank4444** [reply]
Hi, 
Is it supported now?. Since there was a PR to add taproot address generation ([4074](https://github.com/trustwallet/wallet-core/pull/4074))

**Reply by satoshiotomakan** [reply]
Yes, you can now generate P2TR address, and generate and sign transactions spending P2TR UTXOs. Please take a look at the examples:
https://github.com/trustwallet/wallet-core/blob/4af0ee33be559941fbda7d3519a9dd032006ab52/wasm/tests/Blockchain/Bitcoin.test.ts#L10-L93
https://github.com/trustwallet/wallet-core/blob/4af0ee33be559941fbda7d3519a9dd032006ab52/rust/tw_tests/tests/chains/bitcoin/bitcoin_sign/p2tr_key_path.rs
